### PR TITLE
Add blockcopy cases

### PR DESCRIPTION
**Pull request by chloerh**
- with option 'reuse-external'

Signed-off-by: haizhao <haizhao@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/backingchain/blockcopy.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    vm_name = params.get('main_vm')
+    case = params.get('case', '')
+
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    bkxml = vmxml.copy()
+
+    file_to_del = []
+
+    try:
+        if case:
```
Comment (chunfuwen) [inline libvirt/tests/src/backingchain/blockcopy.py]: I assume 'if case' already return True, why need it?

File: libvirt/tests/src/backingchain/blockcopy.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                cur_sfile = cur_disks[0].find('source').get('file')
+                logging.debug('Now disk source file is: %s', cur_sfile)
+                if cur_sfile.strip() != new_image_path:
+                    test.fail('Disk source file is not updated.')
+
+    finally:
+        if case == 'reuse_external':
+            # Recover vm and remove the transient vm
+            virsh.destroy(vm_name, debug=True)
+            virsh.define(bkxml.xml, debug=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/backingchain/blockcopy.py]: this is duplicate with line 75

File: libvirt/tests/src/backingchain/blockcopy.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        bkxml.sync()
+
+        # Remove files to be deleted
+        if file_to_del:
```
Comment (chunfuwen) [inline libvirt/tests/src/backingchain/blockcopy.py]: How can you make use file really exists , which is gotten from file_do_del

File: libvirt/tests/src/backingchain/blockcopy.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcopy.py]: This var is prepared for special cases. There might be other cases that fit in this file in the future, but do not need this var. Maybe some cases with some shared steps that could fit for all cases.

File: libvirt/tests/src/backingchain/blockcopy.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcopy.py]: No, it is not. We're going to remove a transient vm by destroy it(line 73), and recover the original vm(line 74), which are all for case 'reuse_external'.
bkxml.sync() will undefine and redefine, and it's for all cases, not just 'reuse_external'

File: libvirt/tests/src/backingchain/blockcopy.py (hunk 3 above)
Comment (chloerh) [reply libvirt/tests/src/backingchain/blockcopy.py]: Updated. Check if file exists before deleting it.

**Pull request merged**